compiler/src/dotty/tools/dotc/core/TypeApplications.scala

```diff
@@ -172,6 +172,7 @@ class TypeApplications(val self: Type) extends AnyVal {
       case self: TypeRef =>
         val tsym = self.symbol
         if (tsym.isClass) tsym.typeParams
+        else if (!tsym.exists) self.info.typeParams
```

Comment:
.How come we don't have the symbol for the refinement member here? That seems likely to be problematic in a lot of other cases, for example in the IDE how do we do "go to definition" if the tree has no symbol?

Reply:
Refinements of types that do not exist in the baseclass don't have a symbol. Which symbol could they have?